py/desispec/test/test_link_calibnight.py

```diff
@@ -0,0 +1,153 @@
+"""
+test desispec.scripts.link_calibnight
+"""
+
+import os, sys, unittest, tempfile
+from shutil import rmtree
+
+from desispec.io import findfile
+from desispec.io.util import decode_camword
+from desispec.scripts import link_calibnight
+
+class TestLinkCalibNight(unittest.TestCase):
+    """Test desispec.scripts.link_calibnight"""
+
+    @classmethod
+    def setUpClass(cls):
+        #- cache environment variables so that we can reset originals when done
+        cls.cache_env = dict()
+        for name in ['DESI_SPECTRO_REDUX', 'SPECPROD']:
+            cls.cache_env[name] = os.getenv(name)
+
+        #- Setup a test night of calib files
+        cls.testdir = tempfile.mkdtemp()
+        os.environ['DESI_SPECTRO_REDUX'] = cls.testdir
+        os.environ['SPECPROD'] = 'testlinks'
+        cls.reduxdir = f'{cls.testdir}/testlinks'
+        cls.refnight = 20201010
+        cls.altrefnight = 20201020
+        cls.newnight = 20201011
+        os.makedirs(f'{cls.testdir}/testlinks/calibnight/{cls.refnight}')
+        os.makedirs(f'{cls.testdir}/testlinks/calibnight/{cls.altrefnight}')
+        cls.prefixes = ['badcolumns', 'biasnight', 'fiberflatnight', 'psfnight']
+        for night in (cls.refnight, cls.altrefnight):
+            for prefix in cls.prefixes:
+                for camera in decode_camword('a0123456789'):
+                    filename = findfile(prefix, night=night, camera=camera)
+                    with open(filename, 'w') as fx:
+                        fx.write(os.path.basename(filename))
+
+    def tearDown(self):
+        newdir = f'{self.reduxdir}/calibnight/{self.newnight}'
+        if os.path.isdir(newdir):
+            rmtree(newdir)
+
+    @classmethod
+    def tearDownClass(cls):
+        #- remove files
+        rmtree(cls.testdir)
+
+        #- reset environment variables
+        for name in cls.cache_env:
+            if cls.cache_env[name] is None:
+                del os.environ[name]
+            else:
+                os.environ[name] = cls.cache_env[name]
+
+    def test_basic_links(self):
+        options = f'--refnight {self.refnight} --newnight {self.newnight}'.split()
+        link_calibnight.main(options)
+        for prefix in self.prefixes:
```

Comment:
Should you bring in the link_calibnight parser directly and then define `self.prefixes` as `args.include`? Or do you only want to test the subset you define above?

This won't automatically test new filetypes like `cte` unless added explicitly, and would break if we renamed or remove one of the above filenames. Not a major issue but something to consider.

Reply:
As part of argparse updates, I factored out the --include prefix defaults into a `link_calibnight.calibnight_prefixes` module-level variable used by both the argparser and the unit tests (though I still cache that in self.prefixes, mostly as the minimal update to the existing tests).  I think that addresses the concern here, albeit in a different implementation.